service/controller/v3/resource/vpngateway/desired.go

```diff
@@ -0,0 +1,125 @@
+package vpngateway
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2017-09-01/network"
```

Comment:
There is `services/network/mgmt/2018-05-01/network` I think we should use the newest.

Reply:
This imply updating it in all different controller version as they are all bound together by `client` package. There is also some minor changes on peering that need to be addressed.

We can do this in a different PR